### PR TITLE
Blobstore

### PR DESCRIPTION
**Pull request by sunr1**
Set up the Upload Image Servlet that works with the Request Servlet that successfully uploads the user's image to datastore.

### CODE REVIEW

File: bait-files/src/main/java/com/google/sps/servlets/RequestServlet.java [hunk 1]
```diff
@@ -40,24 +49,46 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
...
+    // get user id
     UserService userService = UserServiceFactory.getUserService();
     String userId = userService.getCurrentUser().getUserId();
 
     String nameInput = parameters.get(NAME)[0];
     String usernameInput = parameters.get(USERNAME)[0];
     String emailInput = parameters.get(EMAIL)[0];
     String addressInput = parameters.get(ADDRESS)[0];
-    String pictureInput = parameters.get(PICTURE)[0];
+    String blobkeyString = getUploadedFileUrl(request, PICTURE);
```
Comment (kzweller) [inline bait-files/src/main/java/com/google/sps/servlets/RequestServlet.java:60]: nit: blobKeyString

File: bait-files/src/main/java/com/google/sps/servlets/RequestServlet.java [hunk 2]
```diff
@@ -40,24 +49,46 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
...
         .save()
         .entity(new Request(requestId, userId, nameInput, usernameInput, emailInput, addressInput,
-            pictureInput, phoneInput, notesInput))
+            blobkeyString, phoneInput, notesInput))
         .now();
 
     response.sendRedirect("/submission.html");
   }
+
+  private String getUploadedFileUrl(HttpServletRequest request, String formInputElementName) {
```
Comment (kzweller) [inline bait-files/src/main/java/com/google/sps/servlets/RequestServlet.java:73]: Maybe add a comment somewhere around here that the form restricts the filetype, so we don't need to check here.

File: bait-files/src/main/webapp/blobstore.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
+// Copyright 2019 Google LLC
```
Comment (kzweller) [inline bait-files/src/main/webapp/blobstore.js:1]: This file can be removed right?

File: bait-files/src/main/java/com/google/sps/servlets/UploadImageServlet.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * When the fetch() function requests the /blobstore-upload-url URL, the content of the response is
+ * the URL that allows a user to upload a file to Blobstore. If this sounds confusing, try running a
+ * dev server and navigating to /blobstore-upload-url to see the Blobstore URL.
+ */
+@WebServlet("/upload")
+public class UploadImageServlet extends HttpServlet {
```
Comment (kzweller) [inline bait-files/src/main/java/com/google/sps/servlets/UploadImageServlet.java:17]: Suggest renaming "BlobstoreUploadServlet" and "/blobstore-upload" respectively, or something similar.

File: bait-files/src/test/java/com/google/sps/servlets/RequestServletTest.java [hunk 5]
```diff
@@ -0,0 +1,83 @@
+// package com.google.sps.servlets;
```
Comment (kzweller) [inline bait-files/src/test/java/com/google/sps/servlets/RequestServletTest.java:1]: Is this meant to be a commented out test? What should be happening here?

Comment (sunr1) [reply bait-files/src/test/java/com/google/sps/servlets/RequestServletTest.java:1]: meant for that to be on a different branch - I've deleted it

File: bait-files/src/main/java/com/google/sps/servlets/RequestServlet.java [hunk 6]
```diff
@@ -57,19 +57,24 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     String usernameInput = parameters.get(USERNAME)[0];
     String emailInput = parameters.get(EMAIL)[0];
     String addressInput = parameters.get(ADDRESS)[0];
-    String blobkeyString = getUploadedFileUrl(request, PICTURE);
+    String blobKeyString = getUploadedFileUrl(request, PICTURE);
+
+    BlobKey blobkey = new BlobKey(blobKeyString);
+    blobstoreService.serve(blobkey, response);
```
Comment (kzweller) [inline bait-files/src/main/java/com/google/sps/servlets/RequestServlet.java:63]: Make sure to take these lines out

**Pull request closed without merging**